File: DCodeC#/DCode/Preferences/DCodePrefItemList.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

/**
 * @author Davi
 */

public class DCodePrefItemList {
    protected List<DCodePrefItem> list;
    protected DCode dcode;

    // Constructors
    
    public DCodePrefItemList(DCode dcode, String items){
        This(dcode, ToItemArray(dcode.unCode(items)));
    }
    
    public DCodePrefItemList(DCode dcode, String [] items){
        This(dcode, ToItemArray(items));
    }
    
    public DCodePrefItemList(DCode dcode, DCodePrefItem [] items){
        This(dcode, items);
    }

    // Main constructor
    private void This(DCode dcode, DCodePrefItem [] items) {
        list = ToList(items);
        this.dcode = dcode;
    }

    // Methods

    public void addItem(String key, String value) {
        list.Add(new DCodePrefItem(key, value));
    }

    public DCodePrefItem getItem(String key) {
        DCodePrefItem [] items = list.ToArray();
        for (int i = 0; i < items.Length; i++) {
            if (items [i].getKey().Equals(key)) {
                return items [i];
            }
        }

        return null;
    }

    // The setItem method is on DCodePreferences, becouse uses the getItem();

    public bool contains(String key) {
        return (getItem(key) != null);
    }

    
    // Static methods
    
    public static String [] ToStringArray(DCodePrefItem[] item){
        String [] _item = new String[item.Length];
        for(int i = 0; i < _item.Length; i++){
            _item[i] = item[i].ToString();
        }
        return _item;
    }

    public static String [] ToStringArray(List<DCodePrefItem> items) {
        return ToStringArray(ToItemArray(items));
    }
    
    public static DCodePrefItem[] ToItemArray(String [] items){
        DCodePrefItem [] _items = new DCodePrefItem[items.Length];
        for(int i = 0; i < _items.Length; i++){
            _items[i] = new DCodePrefItem(items[i]);
        }
        return _items;
    }
    
    public static List<DCodePrefItem> ToList(DCodePrefItem [] items){
        List<DCodePrefItem> list = new List<DCodePrefItem>();
        for(int i = 0; i < items.Length; i++){
            list.Add(items[i]);
        }
        return list;
    }
    
    public static DCodePrefItem [] ToItemArray(List<DCodePrefItem> items){
        return items.ToArray();
    }
    
    // Getters and Setters
    
    public List<DCodePrefItem> getList(){
        return list;
    }
    
    // Override methods

    override
    public String ToString() {
        return dcode.enCodeI(ToStringArray(list));
    }
}

File: DCodeJava/src/com/daviapps/dcode/preferences/DCodePrefItemList.java
package com.daviapps.dcode.preferences;

import com.daviapps.dcode.DCode;
import java.util.ArrayList;
import java.util.List;

/**
 * @author Davi, 11/11/2016
 */

public class DCodePrefItemList {
    protected DCode dcode;
    
    // Private variables
    protected List<DCodePrefItem> list;
    
    // Contructors
    
    public DCodePrefItemList(DCode dcode, String items){
        this(dcode, dcode.unCode(items));
    }
    
    public DCodePrefItemList(DCode dcode, String [] items){
        this(dcode, toItemArray(items));
    }
    
    // Main contructor
    public DCodePrefItemList(DCode dcode, DCodePrefItem [] items){
        list = toList(items);
        this.dcode = dcode;
    }
    
    // Methods
    
    public void addItem(String key, String value){
        System.out.println(key + " added with value: " + value);
        list.add(new DCodePrefItem(key, value));
    }
    
    public boolean removeItem(String key){
        for(int i = 0; i < list.size(); i++){
            if(list.get(i).getKey().equals(key)){
                list.remove(i);
                System.out.println(key + " removed");
                return true;
            }
        }
        return false;
    }
    
    public DCodePrefItem getItem(String key){
        for(int i = 0; i < list.size(); i++){
            if(list.get(i).getKey().equals(key)){
                return list.get(i);
            }
        }
        return null;
    }
    
    // The setItem method is on DCodePreferences, becouse uses the getItem();
    
    public boolean contains(String key){
        return (getItem(key) != null);
    }
    
    // Static methods
    
    public static String [] toStringArray(DCodePrefItem[] item){
        String [] _item = new String[item.length];
        for(int i = 0; i < _item.length; i++)
            _item[i] = item[i].toString();
        return _item;
    }
    
    public static String [] toStringArray(List<DCodePrefItem> items){
        return toStringArray(toItemArray(items));
    }
    
    public static DCodePrefItem[] toItemArray(String [] items){
        DCodePrefItem [] _items = new DCodePrefItem[items.length];
        for(int i = 0; i < _items.length; i++)
            _items[i] = new DCodePrefItem(items[i]);
        return _items;
    }
    
    public static List<DCodePrefItem> toList(DCodePrefItem [] items){
        List<DCodePrefItem> list = new ArrayList<>();
        for(int i = 0; i < items.length; i++)
            list.add(items[i]);
        return list;
    }
    
    public static DCodePrefItem [] toItemArray(List<DCodePrefItem> items){
        DCodePrefItem [] out = new DCodePrefItem[items.size()];
        for(int i = 0; i < items.size(); i++)
            out[i] = items.get(i);
        return out;
    }
    
    // Getters and Setters
    
    public List<DCodePrefItem> getList(){
        return list;
    }
    
    // Override methods

    @Override
    public String toString() {
        return dcode.enCode(toStringArray(list));
    }
    
}
